background/services/abilities/index.ts

```diff
@@ -109,30 +110,97 @@ export default class AbilitiesService extends BaseService<Events> {
 
   // Should only be called with ledger or imported accounts
   async getNewAccountAbilities(address: string): Promise<void> {
-    this.pollForAbilities(address)
+    this.pollForAbilities(normalizeEVMAddress(address))
     this.emitter.emit("newAccount", address)
   }
 
-  async pollForAbilities(address: HexString): Promise<void> {
+  async removeAbilities(abilities: Ability[]): Promise<void> {
+    const cachedAbilities = await this.db.getAbilities()
+    const abilitiesById = new Set(abilities.map(({ abilityId }) => abilityId))
+    const diffAbilities = cachedAbilities.filter(
+      (cachedAbility) => !abilitiesById.has(cachedAbility.abilityId)
+    )
+    const removedAbilities = diffAbilities.map((ability) => ({
+      ...ability,
+      removedFromUi: true,
+    }))
+    await this.db.updateAbilities(removedAbilities)
+  }
+
+  async pollForAbilities(address: NormalizedEVMAddress): Promise<void> {
     const daylightAbilities = await getDaylightAbilities(address)
     const normalizedAbilities = normalizeDaylightAbilities(
       daylightAbilities,
       address
     )
+    /**
+     * 1. Remove abilities from the cache
+     * We are not able to get information about the removed abilities from the Daylight API.
+     * To update the cache we have to compare the data with the received abilities.
+     * The ability can be open completed or expired. Therefore, we need to get the abilities for these 3 types.
+     */
+    await this.removeAbilities(normalizedAbilities)
+    /**
+     * 2. Update existing abilities in the cache
+     * We allow users to mark abilities as completed or removed, we do not want to overwrite this state.
+     * There is an exception when the ability is really completed we want to update this property as well.
+     */
+    const cachedAbilities = await this.db.getAbilities()
+    const { updatedAbilitiesByUser, ids } = cachedAbilities.reduce<{
+      updatedAbilitiesByUser: Ability[]
+      ids: Set<string>
+    }>(
+      (acc, ability) => {
+        if (ability.removedFromUi || ability.completed) {
+          acc.ids.add(ability.abilityId)
+          acc.updatedAbilitiesByUser.push(ability)
+        }
+        return acc
+      },
+      { updatedAbilitiesByUser: [], ids: new Set() }
+    )
+    const updatedAbilities = normalizedAbilities.reduce<Ability[]>(
+      (acc, ability) => {
+        if (ids.has(ability.abilityId)) {
+          const existingAbility = updatedAbilitiesByUser.find(
+            ({ abilityId }) => abilityId === ability.abilityId
+          )
+          if (JSON.stringify(ability) !== JSON.stringify(existingAbility)) {
+            const { removedFromUi, completed } = existingAbility ?? {
+              removedFromUi: false,
+              completed: false,
+            }
+            // Update when the ability is really completed but the cache status is not updated
+            const updateCompleted =
+              ability.completed === true && completed === false
```

Comment:
Why we compare booleans like this?
```suggestion
            const updateCompleted = ability.completed && !completed
```